greynoise_connector.py

```diff
@@ -161,438 +148,353 @@ def _is_valid_ip(self, input_ip_address):
+
+                # Extract data and metadata
+                current_data = api_response.get("data", [])
+                request_metadata = api_response.get("request_metadata", {})
+
+                # Check for empty results
+                if request_metadata.get("count", 0) == 0:
+                    message = "No Results Found"
+                    break
+
+                # Update custom message if available
+                if request_metadata.get("message"):
+                    message = request_metadata.get("message")
+
+                # Add current data to results
+                query_response["data"].extend(current_data)
+
+                # Update remaining count
+                if remaining is not None:
+                    remaining -= len(current_data)
+                    if remaining <= 0:
+                        break
+
+                # Get scroll token for next page
+                scroll = request_metadata.get("scroll")
+                if not scroll:
+                    break  # No more pages available
 
         except Exception as e:
-            error_message = self._get_error_message_from_exception(e)
-            if "403" in error_message:
-                results = {"ip": param["ip"], "message": "Not allowed.", "similar_ips": []}
-                action_result.add_data(results)
-                return action_result.set_status(phantom.APP_SUCCESS, "Lookup Similar IPs action not allowed")
-            else:
-                return action_result.set_status(phantom.APP_ERROR, urllib.parse.unquote(error_message))
+            message = f"Error occurred while fetching query details: {self._get_error_message_from_exception(e)}"
+            return phantom.APP_ERROR, query_response, message
 
-        return action_result.set_status(phantom.APP_SUCCESS, "Lookup Similar IPs action successfully completed")
+        return phantom.APP_SUCCESS, query_response, message
 
     def _lookup_ip_timeline(self, param):
         self.save_progress(GREYNOISE_ACTION_HANDLER_MESSAGE.format(identifier=self.get_action_identifier()))
         action_result = self.add_action_result(ActionResult(dict(param)))
-
-        license_type, message = self._check_license_type()
-        if license_type == "community":
-            return action_result.set_status(phantom.APP_ERROR, message)
-
-        session = GreyNoise(api_key=self._api_key, integration_name=self._integration_name)
+        # Extract required parameter
+        ip = param["ip"]
+        field = param.get("field", "classification")
+
+        # Validate field parameter
+        if not self._validate_timeline_field(field, action_result):
+            return action_result
+
+        # Build optional parameters dict
+        api_params = {}
+
+        # Only add parameters if they're provided
+        if "days" in param:
+            api_params["days"] = param["days"]
+        if "granularity" in param:
```

Comment:
Let's provide a default of "1d" for this.

Reply:
Added default value 1d.